src/configuration-requestor.ts

```diff
@@ -41,22 +47,57 @@ export default class ConfigurationRequestor {
         createdAt: configResponse.createdAt,
       });
 
-      // TODO: different polling intervals for bandit parameters
-      const banditResponse = await this.httpClient.getBanditParameters();
-      if (banditResponse?.bandits) {
-        if (!this.banditModelConfigurationStore) {
-          throw new Error('Bandit parameters fetched but no bandit configuration store provided');
-        }
+      if (this.requiresBanditModelConfigurationStoreUpdate(configResponse.banditReferences)) {
+        const banditResponse = await this.httpClient.getBanditParameters();
+        if (banditResponse?.bandits) {
+          if (!this.banditModelConfigurationStore) {
```

Comment:
what do you think about moving this check before `requiresBanditModelConfigurationStoreUpdate`? Then you wouldn't have to make this check again in that method. It would also save having to do a fetch just to throw the error because there's nowhere to store the fetched data.

You could also pass the current loaded bandits as a parameter. I think the methods will be more testable as well if you can replace `this.` references with params.